packages/iov-bcp-types/src/transactions.ts

```diff
@@ -91,40 +87,40 @@ export interface SwapCounterTransaction extends UnsignedTransaction {
 }
 
 export interface SwapClaimTransaction extends UnsignedTransaction {
-  readonly kind: "swap_claim";
+  readonly kind: "bcp/swap_claim";
   readonly preimage: Uint8Array;
   readonly swapId: SwapIdBytes; // pulled from the offer transaction
 }
 
 export interface SwapTimeoutTransaction extends UnsignedTransaction {
-  readonly kind: "swap_timeout";
+  readonly kind: "bcp/swap_timeout";
   readonly swapId: SwapIdBytes; // pulled from the offer transaction
 }
 
 export function isSendTransaction(transaction: UnsignedTransaction): transaction is SendTransaction {
-  return transaction.domain === "bcp" && transaction.kind === "send";
+  return (transaction as SendTransaction).kind === "bcp/send";
```

Comment:
This casts adds type safety for the value to check against. While

```ts
return transaction.kind === "send";
```

compiles, 

```ts
return (transaction as SendTransaction).kind === "send";
```

does not because it is always false. Thus I left the casts there to catch those kind of bugs.

It also follows the general rule (which I still don't like, but it is documented by TS): cast the incoming variable to the type to check for, do some checks that verify the casting was okay.

Reply:
Good explanation. Thanks!